### PR TITLE
Enable multinetting for VLAN and Bond interfaces

### PR DESCRIPTION
**Pull request by hugonicodemos**
This commit enables multi-netting configuration for VLAN and Bond network types as well. Previously, only the Ethernet type was checking the configuration.

Test Plan
[Day 1]
- Add admin multi-netting configuration: no changes should be applied. [Day 2]
- Add admin multi-netting configuration for the 3 types (Ethernet, VLAN, Bond). The configuration should be applied w/o lock/unlock.

### CODE REVIEW

File: controllers/host/host_controller.go [hunk 1]
```diff
@@ -1168,13 +1168,25 @@ func (r *HostReconciler) ReconcileHostByState(client *gophercloud.ServiceClient,
...
-		return nil, false
+		return "", nil, false
 	}
 
+	if ethInfo, hasChange := hasEthernetAdminNetworkChange(profileInterfaces, currentInterfaces); hasChange {
+		return ethInfo.Name, &ethInfo.CommonInterfaceInfo, true
+	} else if VLANInfo, hasChange := hasVLANAdminNetworkChange(profileInterfaces, currentInterfaces); hasChange {
+		return VLANInfo.Name, &VLANInfo.CommonInterfaceInfo, true
+	} else if BondInfo, hasChange := hasBondAdminNetworkChange(profileInterfaces, currentInterfaces); hasChange {
+		return BondInfo.Name, &BondInfo.CommonInterfaceInfo, true
```
Comment (yjian118) [inline controllers/host/host_controller.go:1183]: I saw another IF type "VF" other than "eth" "vlan" "bond", do we need to also check that type?

File: controllers/host/host_controller.go [hunk 2]
```diff
@@ -1187,6 +1199,41 @@ func hasAdminNetworkChange(profileInterfaces *starlingxv1.InterfaceInfo, current
 			return &profileEth, containsAdminPlatformNetwork(*profileEth.PlatformNetworks)
 		}
 	}
+
+	return nil, false
+}
+
+func hasVLANAdminNetworkChange(profileInterfaces, currentInterfaces *starlingxv1.InterfaceInfo) (*starlingxv1.VLANInfo, bool) {
```
Comment (yjian118) [inline controllers/host/host_controller.go:1206]: Can we add simple UTs against these new methods?

File: controllers/host/host_controller.go (hunk 1 above)
Comment (hugonicodemos) [reply controllers/host/host_controller.go:1183]: No, the mgmt or admin network cannot be of the VF type.

File: controllers/host/host_controller.go (hunk 2 above)
Comment (hugonicodemos) [reply controllers/host/host_controller.go:1206]: ack

File: controllers/host/host_controller.go (hunk 1 above)
Comment (yjian118) [reply controllers/host/host_controller.go:1183]: acked

**Review by yjian118 — APPROVED** [review]

**Review by sriram-gn — APPROVED** [review]
LGTM

**Pull request merged**